rosette-scala/src/main/scala/coop/rchain/rosette/Tuple.scala

```diff
@@ -1,14 +1,145 @@
 package coop.rchain.rosette
 
+/** Tuple
+  *
+  *  Tuples are used to represent messages that convey information
+  *  between two actors in a communication, to introduce sub-structure
+  *  in messages, and as a primitive data structure in the definition
+  *  of other actors
+  */
 case class Tuple(elem: Seq[Ob],
-                 override val entry: Seq[Ob],
+                 override val parent: Ob,
                  override val meta: Ob,
                  override val slot: Seq[Ob])
-    extends Ob
+    extends Ob {
+
+  def makeSlice(offset: Int, n: Int): Tuple =
+    this match {
+      case Tuple.NIL => Tuple.NIL
+      case _ => Tuple(n, this, offset, n)
+    }
+
+  def makeTail(entriesToSkip: Int): Tuple = {
+    val size = this.elem.size - entriesToSkip
```

Comment:
size could end up negative... is that by design?

Here's hoping I find time to do some property testing...

Reply:
If we skip more entries than we have and `size` ends up negative, we end up with an empty `Tuple`.
I think this is fine?